libultraship/libultraship/Archive.cpp

```diff
@@ -55,15 +66,19 @@ namespace Ship {
 		}
 	}
 
-	std::shared_ptr<File> Archive::LoadFile(const std::string& filePath, bool includeParent, std::shared_ptr<File> FileToLoad) {
+	std::shared_ptr<File> Archive::LoadFile(const std::string& filePath, bool includeParent, std::shared_ptr<File> FileToLoad, HANDLE mpqHandle) {
```

Comment:
Why pass in mpq handle here?

Reply:
Previously LoadFile would only ever load from mainMpq, I added this so that PushGameVersion can specify which MPQ file it gets the version file from a different MPQ file if needed. The workflow is, after loading the Main MPQ file, it loads all the other OTRs as patches via `LoadPatchMPQ`. `LoadPatchMPQ` opens the archive first with it's own `mpqHandle` and then checks it's version file and pushes that to the list of rom hashes that were loaded. So it passes the `patchMpqHandle` to `PushGameVersion` which then pushes it to `LoadFile`. I believe I have it set up so that if you don't provide an `mpqHandle` it will just use the main mpq handle.

Reply:
Do we need to read from a specific otr when all are loaded as patches? To get the master quest version of a scene you just change the string reference. The patching system should take care of what data is loaded, not passing in a handle. If you need to change out OTRs the the archive needs to be reloaded.

Reply:
Well this functionality is only ever used to read the version file _before_ patching so it can store it in a set that any game using libultraship can read later. It's set up with defaults such that you can call LoadFile without providing a handle and it will load from the mainMPQ handle just like before. If you really want it gone I can just call the stormlib functions directly in `PushGameVersion` instead, I was just avoiding code duplication and needed a way to load from a different OTR for that particular case. I didn't think the option to load from a specific mpqHandle would be unwelcome.

Reply:
If the intention is to load the version file then we need a private overloaded method that takes in the handle. The public one calls the private one with the handle as null.

Reply:
Works for me, done in latest commit.